### PR TITLE
refactor: upgrade kube-builder apiversion and project layout

### PR DESCRIPTION
**Pull request by wawlian**
#### What type of PR is this?
/kind enhancement

#### What this PR does / why we need it:
1. upgrade kube-builder version to v3
2. convert kube-builder project layout to multi group
3. add pkg for controller and client logic
4. change data type of DeviceResource.Attributes (DeviceProfile) to map[string]string

#### Special notes for your reviewer:
/assign @gnunu @qclc 

#### Does this PR introduce a user-facing change?
Yes, the data type of DeviceResource.Attributes (DeviceProfile) to map[string]string to fix make manifests error.

### CODE REVIEW

File: pkg/controllers/device_syncer.go [hunk 1]
```diff
@@ -21,12 +21,13 @@ import (
...
 	"time"
 
-	devicev1alpha1 "github.com/openyurtio/device-controller/api/v1alpha1"
-	edgeCli "github.com/openyurtio/device-controller/clients"
-	efCli "github.com/openyurtio/device-controller/clients/edgex-foundry"
-	"github.com/openyurtio/device-controller/cmd/yurt-device-controller/options"
-	"github.com/openyurtio/device-controller/controllers/util"
+	edgeCli "github.com/openyurtio/device-controller/pkg/clients"
+	efCli "github.com/openyurtio/device-controller/pkg/clients/edgex-foundry"
+	util2 "github.com/openyurtio/device-controller/pkg/controllers/util"
```
Comment (gnunu) [inline pkg/controllers/device_syncer.go:26]: util2 -> util?

File: pkg/controllers/deviceprofile_syncer.go [hunk 2]
```diff
@@ -21,16 +21,18 @@ import (
 	"strings"
 	"time"
 
+	devcli "github.com/openyurtio/device-controller/pkg/clients"
+	edgexclis "github.com/openyurtio/device-controller/pkg/clients/edgex-foundry"
+	util2 "github.com/openyurtio/device-controller/pkg/controllers/util"
```
Comment (gnunu) [inline pkg/controllers/deviceprofile_syncer.go:26]: util2 -> util?

File: pkg/controllers/util/fieldindexer.go [hunk 3]
```diff
@@ -20,9 +20,9 @@ import (
 	"context"
 	"sync"
 
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"github.com/openyurtio/device-controller/apis/device.openyurt.io/v1alpha1"
```
Comment (gnunu) [inline pkg/controllers/util/fieldindexer.go:23]: v1alpha1 -> devicev1alpha1?

File: pkg/controllers/util/tools.go [hunk 4]
```diff
@@ -22,11 +22,11 @@ import (
 	"io/ioutil"
 	"strings"
 
+	"github.com/openyurtio/device-controller/apis/device.openyurt.io/v1alpha1"
```
Comment (gnunu) [inline pkg/controllers/util/tools.go:25]: v1alpha1 -> devicev1alphav1

**Review by gnunu — DISMISSED** [review]

**Review by gnunu — APPROVED** [review]

**Review by qclc — COMMENTED** [review]
This modification involves a large number of import package path modifications, but the format is a bit confusing, please organize the import package into the following format
```
import (
      built-in packages in golang

      openyurtio related packages

      third-party packages
)
```

**Review by qclc — COMMENTED** [review]
/approve

**Pull request merged**